### PR TITLE
fix: flux regression in buckets query

### PR DESCRIPTION
**Pull request by lesam**
Closes https://github.com/influxdata/influxdb/issues/21184
Closes https://github.com/influxdata/chronograf/issues/5721

Update the buckets source for the latest flux version

- [x] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass

### CODE REVIEW

File: flux/stdlib/influxdata/influxdb/buckets.go [hunk 1]
```diff
@@ -16,8 +16,23 @@ import (
 	"github.com/influxdata/influxql"
 )
 
+const BucketsKind = "influxdata/influxdb.localBuckets"
+
 func init() {
-	execute.RegisterSource(influxdb.BucketsKind, createBucketsSource)
+	execute.RegisterSource(BucketsKind, createBucketsSource)
+	plan.RegisterPhysicalRules(LocalBucketsRule{})
```
Comment (lesam) [inline flux/stdlib/influxdata/influxdb/buckets.go:23]: Previously registering the `source` was good enough, but after the flux upgrade we also need to register a rule to rewrite a remote buckets call with no host to a local buckets call.

File: flux/stdlib/influxdata/influxdb/buckets.go [hunk 2]
```diff
@@ -126,3 +141,27 @@ type MetaClient interface {
...
+
+func (rule LocalBucketsRule) Name() string {
+	return "influxdata/influxdb.LocalBucketsRule"
+}
+
+func (rule LocalBucketsRule) Pattern() plan.Pattern {
+	return plan.Pat(influxdb.BucketsKind)
+}
+
+func (rule LocalBucketsRule) Rewrite(ctx context.Context, node plan.Node) (plan.Node, bool, error) {
```
Comment (davidby-influx) [inline flux/stdlib/influxdata/influxdb/buckets.go:155]: Consider naming the return values for clarity.

**Review by davidby-influx — APPROVED** [review]
LGTM. One non-blocking comment.

File: flux/stdlib/influxdata/influxdb/buckets.go (hunk 2 above)
Comment (lesam) [reply flux/stdlib/influxdata/influxdb/buckets.go:155]: Copied from 2.x so will keep for now

**Pull request merged**